src/provider/file/mod.rs

```diff
@@ -602,22 +575,26 @@ impl PartFile {
         // Verify parcel by rewinding the parcel and then hashing it.
         // This MUST be after the last write to out, otherwise the results will
         // not be correct.
-        out.flush().await?;
-        out.seek(std::io::SeekFrom::Start(0)).await?;
+        self.file.flush().await?;
+        self.file.seek(std::io::SeekFrom::Start(0)).await?;
         trace!("Validating data for parcel");
-        validate_sha256(&mut out, parcel_id)
+        validate_sha256(&mut self.file, parcel_id)
             .instrument(tracing::trace_span!("parcel_data_validation"))
             .await?;
         trace!("SHA data validated");
         Ok(())
     }
 
     /// Moves the file to the configured final location, consuming the part file
-    async fn finalize(self) -> Result<()> {
+    async fn finalize(mut self) -> Result<()> {
         debug!(
             renamed_path = %self.final_location.display(),
             "Renaming part file for parcel"
         );
+
+        // Close the file handle to avoid any problems with unfinished IO operations
+        self.file.shutdown().await?;
```

Comment:
Did the previous version not close the file here?

Reply:
In this PR I added the file, before this PR it was dropped inside of the `write_data` function, which cleans it up. But because I am doing a `rename` on it, I didn't want any problems because I still had the file handle around